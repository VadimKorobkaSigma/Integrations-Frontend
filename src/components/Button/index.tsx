import * as React from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
    variant?: 'primary' | 'link';
}

const Button: React.FC<Props> = ({ children, ...args }) => {
    return (
        <button {...args} className={cn(styles.button, styles[args.variant || 'primary'], args.className)}>
            {children}
        </button>
    );
};

export default React.memo(Button);
