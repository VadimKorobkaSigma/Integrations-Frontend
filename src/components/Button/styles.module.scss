.button {
  color: #fff;
  background-color: #2991f3;
  margin-bottom: 1rem;
  width: 160px;
  border: 0;
  padding: 0.5rem 1rem;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.15);
  border-radius: 4px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  font-size: 20px;

  &:hover {
    background-color: #1579d7;
  }
}
