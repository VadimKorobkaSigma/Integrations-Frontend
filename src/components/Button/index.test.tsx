import * as React from 'react';
import Button from './index';
import { shallow } from 'enzyme';

test('Button Component with children', () => {
    const wrapper = shallow(<Button>Submit</Button>);
    expect(wrapper.html()).toMatchSnapshot();
    const onClick = jest.fn();
    wrapper.setProps({ onClick });
    wrapper.find('button').simulate('click');
    expect(onClick).toBeCalledTimes(1);
});
