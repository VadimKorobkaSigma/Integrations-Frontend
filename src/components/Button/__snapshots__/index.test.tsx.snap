// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component with children 1`] = `"<button class=\\"\\">Submit</button>"`;
